firedrake/preconditioners/assembled.py

```diff
@@ -72,3 +94,12 @@ def view(self, pc, viewer=None):
         if hasattr(self, "pc"):
             viewer.printfASCII("PC to apply inverse\n")
             self.pc.view(viewer)
+
+
+class ExplicitSchurPC(AssembledPC):
+
+    _prefix = "schur_"
+
+    def form(self, test, trial, pc):
+        """Return (a, bcs)"""
+        raise NotImplementedError
```

Comment:
Prefer

```python
import abc
class ExplicitSchurPC(AssembledPC, metaclass=abc.ABCMeta):

    @abstractmethod
    def form(self, test, trial, pc):
        ....
```

As well, would it make sense to mimic all the other methods for these classes and reorder so that the signature is:

```python
def form(self, pc, test, trial)
```

Can you also elaborate in the docstring on what the arguments are?

Arguably, the name is a little off, in that this doesn't really have anything to do with Schur complements.  I can't think of a better one straight away though.

Reply:
I've added the `abc.abstractmethod` stuff, changed the arguments order, and added more complete docstrings.